addons/hr_holidays/models/hr_leave_type.py

```diff
@@ -602,3 +345,194 @@ def action_see_accrual_plans(self):
+                if len(allocations_leaves_consumed[employee][leave_type].items()) == 0:
+                    continue
+                lt_info = (
+                    leave_type.name,
+                    {
+                        'remaining_leaves': 0,
+                        'virtual_remaining_leaves': 0,
+                        'max_leaves': 0,
+                        'accrual_bonus': 0,
+                        'leaves_taken': 0,
+                        'virtual_leaves_taken': 0,
+                        'leaves_requested': 0,
+                        'leaves_approved': 0,
+                        'closest_allocation_remaining': 0,
+                        'closest_allocation_expire': False,
+                        'request_unit': leave_type.request_unit,
+                        'icon': leave_type.sudo().icon_id.url,
+                    },
+                    leave_type.requires_allocation,
+                    leave_type.id)
+                leave_allocations = self.env['hr.leave.allocation']
+                for allocation, data in allocations_leaves_consumed[employee][leave_type].items():
+                    future_leaves = 0
+                    if allocation.allocation_type == 'accrual':
+                        future_leaves = allocation.get_future_leaves_on(date)
+                    if not allocation or (not allocation.date_to or allocation.date_to >= date) and allocation.date_from <= date:
+                        lt_info[1]['remaining_leaves'] += data['remaining_leaves'] + future_leaves
+                        lt_info[1]['virtual_remaining_leaves'] += data['virtual_remaining_leaves'] + future_leaves
+                        lt_info[1]['max_leaves'] += data['max_leaves'] + future_leaves
+                        lt_info[1]['accrual_bonus'] += future_leaves
+                        lt_info[1]['leaves_taken'] += data['leaves_taken']
+                        lt_info[1]['virtual_leaves_taken'] += data['virtual_leaves_taken']
+                        lt_info[1]['leaves_requested'] += data['virtual_leaves_taken']
+                        lt_info[1]['leaves_approved'] += data['leaves_taken']
+                        leave_allocations |= allocation
+                sorted_leave_allocations = leave_allocations.filtered('date_to').sorted(key='date_to')\
+                    + leave_allocations.filtered(lambda a: not a.date_to)\
+                    if leave_allocations else False
+                closest_allocation = sorted_leave_allocations[0] if sorted_leave_allocations else self.env['hr.leave.allocation']
+                closest_allocation_remaining = allocations_leaves_consumed[employee][leave_type][closest_allocation]['virtual_remaining_leaves']\
+                    if closest_allocation else 0
+                if closest_allocation.date_to:
+                    closest_allocation_expire = format_date(self.env, closest_allocation.date_to, date_format="MM/dd/yyyy")
+                    closest_allocation_duration =\
```

Comment:
don't get what is closest_allocation_duration :thinking: 

update: okey, I see now.. but we could also put a comment to explain what it is.. :innocent: